### PR TITLE
PR for Security Critical Permissions

### PR DESCRIPTION
**Pull request by MatthijsKrempel**
As mentioned in; https://github.com/OrchardCMS/OrchardCore/issues/2177

I've decorated the permissions with a boolean to indicate if the permission is security critical.

I've also added a badge next to the permission in the roles screen so the user will be made aware of this.

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Roles/Views/Admin/Edit.cshtml [hunk 1]
```diff
@@ -41,6 +51,7 @@
                                         <label class="custom-control-label" for="Effective.@permission.Name">&nbsp;</label>
                                     </div>
                                 </td>
+
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Roles/Views/Admin/Edit.cshtml]: Remove this line

File: src/OrchardCore.Modules/OrchardCore.Roles/Views/Admin/Edit.cshtml [hunk 2]
```diff
@@ -58,7 +69,7 @@
 </form>
 
 <script at="Foot" type="text/javascript">
-//<![CDATA[
+    //<![CDATA[
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Roles/Views/Admin/Edit.cshtml]: Remove the CDATA in your way while it's not used at all

**Comment by MatthijsKrempel** [comment]
I think we can display a dialog when the maintainer activates this permission, but I think a sticker with WARNING is good enough to solve the problem.

**Comment by hishamco** [comment]
@sebastienros your thoughts, while you are the one who filed this issue

**Comment by sebastienros** [comment]
- text needs to be translated
- a new column is not necessary, just add an icon in the same column
- can the icon have a title such that the tooltip would explain what the warning is about?
- Maybe check other potential permissions that should have this (run workflows, run recipes)

**Comment by hishamco** [comment]
> can the icon have a title such that the tooltip would explain what the warning is about?

I saw similar idea in dot.net
Tooltip

**Comment by sebastienros** [comment]
@hishamco that's the idea. Question for you, would a `pull-right` work in the column for RTL ?

**Comment by hishamco** [comment]
> can the icon have a title such that the tooltip would explain what the warning is about?

I saw similar idea in dot.net
Tooltip

**Comment by hishamco** [comment]
@MatthijsKrempel if you need any kind of help in the UI, please ping me

> Question for you, would a pull-right work in the column for RTL ?

@sebastienros `pull-right` is used for the previous version of bootstrap, now it's replace with `float-right`, let me check this one on RTL

**Comment by MatthijsKrempel** [comment]
@sebastienros @hishamco will get on it.

File: src/OrchardCore.Modules/OrchardCore.Workflows/Permissions.cs [hunk 3]
```diff
@@ -8,7 +8,7 @@ namespace OrchardCore.Workflows
     public class Permissions : IPermissionProvider
     {
         public static readonly Permission ManageWorkflows = new Permission("ManageWorkflows", "Manage workflows");
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Workflows/Permissions.cs]: This one too, as you can edit/create one with it

**Pull request merged**